src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2346,7 +2347,26 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False):  #
         _create_new_asp = True
     else:
         logger.warning("Resource group '%s' already exists.", rg_name)
-        _create_new_asp = should_create_new_asp(cmd, rg_name, asp, location)
+        _asp_generic = asp[:-len(asp.split("_")[4])]
+        # get all asp in the RG
+        data = (list(filter(lambda x: _asp_generic in x.name,
+                            client.app_service_plans.list_by_resource_group(rg_name))))
+        data_sorted = (sorted(data, key=lambda x: x.name))
+        num_asps = len(data)
+        # check if any of these matches the SKU & location to be used
+        # and get FirstOrDefault
+        selected_asp = next((a for a in data if isinstance(a.sku, SkuDescription) and
+                             a.sku.tier.lower() == full_sku.lower() and
+                             (a.location.replace(" ", "").lower() == location or a.location == location)), None)
+        if selected_asp is not None:
+            asp = selected_asp.name
+            _create_new_asp = False
+        elif selected_asp is None and num_asps > 0:
+            # from the sorted data pick the last one & check if a new ASP needs to be created
+            # based on SKU or not
+            _plan_info = data_sorted[num_asps - 1]
+            _asp_num = int(_plan_info.name.split('_')[4]) + 1
+            asp = "{}_asp_{}_{}_{}".format(user, os_val, loc_name, _asp_num)
```

Comment:
same here, if the name are for user to read, then make it nice like `name`, `name_2`, etc